formats/onenote.ts

```diff
@@ -0,0 +1,345 @@
+import { FormatImporter } from 'format-importer';
+import { ProgressReporter } from 'main';
+import { DataWriteOptions, Notice, Setting, TFile, TFolder, htmlToMarkdown, requestUrl } from 'obsidian';
+import { OnenotePage, OnenoteSection, Notebook, SectionGroup } from '@microsoft/microsoft-graph-types';
+import { parseHTML } from '../util';
+import { DeviceCode, TokenResponse } from './onenote/models/device-code';
+
+const GRAPH_CLIENT_ID: string = 'c1a20926-78a8-47c8-a2a4-650e482bd8d2'; // TODO: replace with an Obsidian team owned client_Id
+const GRAPH_SCOPES: string[] = ['user.read', 'notes.read'];
+const ATTACHMENT_EXTS: string[] = ['png','webp','jpg','jpeg','gif','bmp','svg','mpg','m4a','webm','wav','ogv','3gp','mov','mp4','mkv','pdf'];
+
+export class OneNoteImporter extends FormatImporter {
+	attachmentQueue: {name: string, url: string } [] = [];
+	requestParams: RequestInit;
+	useDefaultAttachmentFolder: boolean;
+	importIncompatibleAttachments: boolean;
+	selectedSections: OnenoteSection[] = [];
+
+	init() {
+		this.addOutputLocationSetting('OneNote');
+
+		this.showUI();
+	}
+
+	async showUI() {
+		new Setting(this.modal.contentEl)
+			.setName('Use the default attachment folder')
+			.setDesc('If disabled, attachments will be stored in the export folder in the OneNote Attachments folder.')
+			.addToggle((toggle) => {
+				toggle.setValue(true).onChange((value) => (this.useDefaultAttachmentFolder = value));
+			});
+		new Setting(this.modal.contentEl)
+			.setName('Import incompatible attachments')
+			.setDesc('Imports incompatible attachments which cannot be embedded in Obsidian, such as .exe files.')
+			.addToggle((toggle) => {
+				toggle.setValue(false);
+				toggle.onChange((value) => (this.importIncompatibleAttachments = value));
+			});
+
+		// Create a wrapper for sign in related settings in order to hide them later
+		const contentArea = this.modal.contentEl.createDiv({
+			cls: 'contentArea',
+		});
+		contentArea.createEl('h3', {
+			text: 'Sign in to your Microsoft Account',
+			cls: 'modal-title',
+		});
+		let description = contentArea.createEl('p');
+		// This could possibly use the version from DeviceCode.message, as it returns a string in the user's language?
+		description.innerHTML = `Go to <a href="https://microsoft.com/devicelogin">microsoft.com/devicelogin</a> on your PC or phone and enter this code: <b>${await this.generateLoginCode()}</b>`;
+
+		new Setting(contentArea)
+			.setName('Custom user access token')
+			.setDesc('If you are having troubles with the device code, use a custom access token from Microsoft Graph Explorer by going to the access token tab under the address bar.')
+			.addText((text) => text.setPlaceholder('Paste token here').onChange(async (e) => this.signIn(e)));
```

Comment:
nit: it looks more clear if you write `.setPlaceholder` and it's own line and then `.onChange` on another line